### DISCUSSION TITLE
Self bot implementation

### DISCUSSION THREAD

**Discussion by Spaceish** [Kirdock/discordjs-voice-recorder#3] [Ideas]
It would be cool if you would be able to record conversations from your discord account, like in group chats

**Comment by Kirdock** [comment]
Could you explain it a bit further? You can use the lib to get a track per user.
Oh, do you mean a feature to only get a single user track instead of all user tracks?

**Reply by Spaceish** [reply]
No, like this is only for bots I suppose.I want an implementation to also make it work on user accounts, like to record users in a group call. Like make use of  [discord.js-selfbot-v13](https://github.com/aiko-chan-ai/discord.js-selfbot-v13)

**Reply by Kirdock** [reply]
If you can get your voice connection that has the same functionality as the default (bot) one, you can just provide it to the lib and it records the stuff.
The important thing needed is `connection.receiver`
https://github.com/Kirdock/discordjs-voice-recorder/blob/cbf4d3f6a358cf328d6a1c243fc0afccd44629e7/src/voice-recorder.ts#L68
https://github.com/Kirdock/discordjs-voice-recorder/blob/cbf4d3f6a358cf328d6a1c243fc0afccd44629e7/src/voice-recorder.ts#L55
https://github.com/Kirdock/discordjs-voice-recorder/blob/cbf4d3f6a358cf328d6a1c243fc0afccd44629e7/src/voice-recorder.ts#L64

**Reply by Spaceish** [reply]
Ok, thanks for your advice.

**Reply by Spaceish** [reply]
Hello, sorry for bothering you again, but I seem to have a problem with my code:
```javascript
import { VoiceRecorder } from "@kirdock/discordjs-voice-recorder"

import { Client } from "discord.js-selfbot-v13"

import { joinVoiceChannel } from "@discordjs/voice"



import { readFile } from "fs/promises"
import { createWriteStream } from "fs"

function content(path) {
    return readFile(path, 'utf-8')
}

const client = new Client({
    patchVoice: true
})

const tok = await content("token.txt")

const channelID = await content("channelID.txt")

client.on('ready', async () => {
    const vrc = new VoiceRecorder(client)

    const connection = joinVoiceChannel({
        guildId: channelID,
        channelId: channelID,
        adapterCreator: (await client.channels.fetch(channelID)).voiceAdapterCreator,
        selfDeaf: false,
        selfMute: true,
    })

    vrc.startRecording(connection)

    await vrc.getRecordedVoice(
        createWriteStream('recorded_session.mp3'),
        channelID,
        'single',
        1
    )
})

client.login(tok)
```
It seems like the account it's not joining the group call and it immediately stops the recording.

**Reply by Kirdock** [reply]
Of course it immediately stops recording, because you are calling `await vrc.getRecordedVoice` right after `vrc.startRecording`. It saves the last x minutes and because you just started it, there aren't any
